cfg4k-core/src/main/kotlin/com/jdiazcano/cfg4k/reloadstrategies/TimedReloadStrategy.kt

```diff
@@ -17,25 +17,47 @@
 package com.jdiazcano.cfg4k.reloadstrategies
 
 import com.jdiazcano.cfg4k.providers.ConfigProvider
-import java.util.Timer
+import com.jdiazcano.cfg4k.reloadstrategies.TimedReloadStrategy.Mode.FIXED_DELAY
+import com.jdiazcano.cfg4k.reloadstrategies.TimedReloadStrategy.Mode.FIXED_RATE
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.ScheduledFuture
 import java.util.concurrent.TimeUnit
-import kotlin.concurrent.timer
 
 /**
  * This reload strategy will reload the config provider when the timer ticks. A TimeUnit is used which will be
  * translated to milliseconds.
  */
-class TimedReloadStrategy(val time: Long, val unit: TimeUnit) : ReloadStrategy {
+class TimedReloadStrategy(private val time: Long,
+                          private val unit: TimeUnit,
+                          private val mode: Mode = FIXED_RATE) : ReloadStrategy {
+    enum class Mode {
+        FIXED_RATE, FIXED_DELAY
+    }
 
-    private lateinit var reloadTimer: Timer
+    private val executor: ScheduledExecutorService by lazy {
+        Executors.newSingleThreadScheduledExecutor { runnable ->
+            Thread(runnable, "TimeReloadStrategy").also { it.isDaemon = true }
+        }
+    }
+    private val reloadTasks = mutableMapOf<ConfigProvider, ScheduledFuture<*>>()
 
     override fun register(configProvider: ConfigProvider) {
-        reloadTimer = timer("TimeReloadStrategy", true, unit.toMillis(time), unit.toMillis(time)) {
-            configProvider.reload()
+        reloadTasks.computeIfAbsent(configProvider) { cp ->
+            val reload = {
+                try {
+                    cp.reload()
+                } catch (ignored: Exception) {
```

Comment:
Nothing good in swallowing the exception here, but since the rest of the `cfg4k` doesn't log errors, I decided to leave this as is.
I understand that additional dependencies are undesirable, but I can make use of JUL as a workaround ¯\\\_(ツ)\_/¯ Let me know what you think about this.

Reply:
The config provider must handle the exceptions so I would say there is no need to have this try catch here. 

As for the exceptiosn there is an `addReloadErrorListener()` which these listeners will be called when an exception is thrown when loading/reloading.

* [addReloadErrorListener interface](https://github.com/jdiazcano/cfg4k/blob/master/cfg4k-core/src/main/kotlin/com/jdiazcano/cfg4k/providers/ConfigProvider.kt#L85)
* [DefaultConfigProvider notify errors](https://github.com/jdiazcano/cfg4k/blob/master/cfg4k-core/src/main/kotlin/com/jdiazcano/cfg4k/providers/DefaultConfigProvider.kt#L129)

Reply:
Hm, so this is a recent work as I can see, it has not been included into any release so far.
Sorry, I found the issues I tried to address here while working on a pet project with cfg4k added as a dependency, so I was mostly looking into source of version 0.8 and then ported my change and added tests in the actual cfg4k repo.

Cool, so if `ConfigProvider` now has a contract of catching exceptions on reload and notifying possible listeners, then my change is indeed redundant.

Reply:
There's no release yet but I could publish `0.8.1` once this is merged.